### PR TITLE
Config: Change testing library to Vitest from Jest.

### PR DESCRIPTION
**Pull request by Akamig**
### Changes
- Replace Jest to Vitest
  - Interop vitest.config.ts from jest.config.ts
  - Remove Jest Global "TextEncoder" definition, jsdom has been not supporting TextEncoder properly for years now, but happy-dom doesn't have such issue.
 
- Disable yarn 3 pnp
  - Yarn 3 has different pnp directory than 2, so added them in .gitignore

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,15 +91,15 @@
...
-    "babel-jest": "^29.7.0",
+    "@vitest/coverage-v8": "^1.1.3",
     "codecov": "^3.8.3",
     "eslint": "8.42.0",
     "eslint-config-next": "13.4.5",
     "eslint-plugin-react-hooks": "^4.6.0",
-    "jest": "^29.7.0",
-    "jest-environment-jsdom": "^29.7.0",
-    "ts-jest": "^29.1.1",
+    "jsdom": "^23.2.0",
```
Comment (rohjs) [inline package.json]: happy-dom please

File: tsconfig.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "jsx": "preserve",
     "incremental": true
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "jest.config.ts"],
+  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "vitest.config.ts"],
   "exclude": ["node_modules"],
-  "types": ["node", "jest", "@testing-library/jest-dom"]
+  "types": ["node", "vitest/globals", "jsdom"]
```
Comment (rohjs) [inline tsconfig.json]: happy-dom please

**Review by rohjs — CHANGES_REQUESTED** [review]
LGTM except `jsdom` 👍

File: package.json (hunk 1 above)
Comment (Akamig) [reply package.json]: yes m'lord

**Comment by Akamig** [comment]
Codecov action is running yml in main branch, resulting failure. will be retested after merge.

**Comment by rohjs** [comment]
image

**Review by rohjs — APPROVED** [review]

**Pull request merged**